sdk/go/sse.go

```diff
@@ -10,18 +10,69 @@ import (
 	"sync"
 	"time"
 
+	"github.com/CAFxX/httpcompression/contrib/andybalholm/brotli"
+	"github.com/CAFxX/httpcompression/contrib/compress/gzip"
+	"github.com/CAFxX/httpcompression/contrib/compress/zlib"
+	"github.com/CAFxX/httpcompression/contrib/klauspost/zstd"
 	"github.com/valyala/bytebufferpool"
 )
 
 type ServerSentEventGenerator struct {
 	ctx             context.Context
 	mu              *sync.Mutex
-	w               http.ResponseWriter
+	w               io.Writer
 	rc              *http.ResponseController
 	shouldLogPanics bool
+	encoding        string
 }
 
-func NewSSE(w http.ResponseWriter, r *http.Request) *ServerSentEventGenerator {
+type SSEOption func(*ServerSentEventGenerator)
+
+func WithDeflate(level int) SSEOption {
+	return func(sse *ServerSentEventGenerator) {
+		comp, _ := zlib.New(zlib.Options{Level: level})
+		sse.w = comp.Get(sse.w)
+		sse.encoding = zlib.Encoding
+	}
+}
+
+func WithGzip(level int) SSEOption {
+	return func(sse *ServerSentEventGenerator) {
+		comp, _ := gzip.New(gzip.Options{Level: level})
+		sse.w = comp.Get(sse.w)
+		sse.encoding = gzip.Encoding
+	}
+}
+
+func WithBrotli(level int) SSEOption {
+	return func(sse *ServerSentEventGenerator) {
+		comp, _ := brotli.New(brotli.Options{Quality: level})
+		sse.w = comp.Get(sse.w)
+		sse.encoding = brotli.Encoding
+	}
+}
+
+func WithZstd() SSEOption {
+	return func(sse *ServerSentEventGenerator) {
+		comp, _ := zstd.New()
+		sse.w = comp.Get(sse.w)
+		sse.encoding = zstd.Encoding
+	}
+}
+
+func WithDefaultGzip() SSEOption {
```

Comment:
Closer!  But these should take options as well... you should be able to config, not just support the default levels